### PR TITLE
Post about using custom machine hooks

### PR DESCRIPTION
**Pull request by kevinmaes**
Preview Deployment: Append `/blog/just-use-hooks-xstate-in-react-components` to the latest Vercel deployment URL.

Example:

`https://blog-c3el1es5p-statelyai.vercel.app/blog/just-use-hooks-xstate-in-react-components`

### CODE REVIEW

File: content/posts/2022-07-18-just-use-hooks-xstate-in-react-components.mdx [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+publishedAt: 2022-7-18
+excerpt: "XState can be easily added to React components by using custom machine hooks for greater encapsulation and reusability."
+ogImage: "https://stately.ai/blog/2022-07-18-just-use-hooks-xstate-in-react-components.png"
+---
+
+Are you a React developer using XState to model your application logic? Perhaps you’ve heard of XState but have been looking for an easy way to try it out in one of your projects. If so, then I’d like to share with you a pattern I was introduced to when first diving into codebase at Stately, that of using custom machine hooks. This lightweight, reusable way to integrate XState into React components is a delight to work with and I think you might like it as much as I do!
+
+### Introduction
+
+In this post I’ll review the most common way to use `XState/react` library in a project. I’ll then demonstrate how encapsulation and reuse of state machines can be achieved by using hooks in your components, with some examples. I’ll also touch on the advantages and disadvantages to using this level of abstraction.
```
Comment (davidkpiano) [inline content/posts/2022-07-18-just-use-hooks-xstate-in-react-components.mdx:25]:
```suggestion
In this post I’ll review the most common way to use the `@xstate/react` library in a project. I’ll then demonstrate how encapsulation and reuse of state machines can be achieved by using hooks in your components, with some examples. I’ll also touch on the advantages and disadvantages to using this level of abstraction.
```

File: content/posts/2022-07-18-just-use-hooks-xstate-in-react-components.mdx [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+For more background, you can check out [“Just Use Props”: An opinionated guide to React and XState](https://dev.to/mattpocockuk/just-use-props-an-opinionated-guide-to-react-and-xstate-fc9) by Matt Pocock.
+
+After years of usage in the wild, and in response to confusion and frustration about hooks, the React Team has been putting a lot of effort into making the use of hooks clearer and simpler. Now is the perfect time to re-explore how hooks, when used effectively, can help make component creation easier.
+
+### XState in React components
+
+For those of you who are already using XState with React, you’re probably already used to creating a machine using `createMachine()` and then passing that machine to the `useMachine` hook from within a component.
+
+As a reference, here is the [Quick Start example](https://xstate.js.org/docs/packages/xstate-react/#quick-start) from the `xstate/react` docs, where a `toggleMachine` is created and then passed to `useMachine` for use in a component.
```
Comment (davidkpiano) [inline content/posts/2022-07-18-just-use-hooks-xstate-in-react-components.mdx:35]:
```suggestion
As a reference, here is the [Quick Start example](https://xstate.js.org/docs/packages/xstate-react/#quick-start) from the `@xstate/react` docs, where a `toggleMachine` is created and then passed to `useMachine` for use in a component.
```

File: content/posts/2022-07-18-just-use-hooks-xstate-in-react-components.mdx [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+![The example from XState/react documentation that shows a toggle machine being created with the createMachine function and that machine passed to the useMachine hook.](2022-07-18-useMachine-hook-example.png)
```
Comment (davidkpiano) [inline content/posts/2022-07-18-just-use-hooks-xstate-in-react-components.mdx:37]: Can we copy the actual code from the documentation?

Comment (kevinmaes) [reply content/posts/2022-07-18-just-use-hooks-xstate-in-react-components.mdx:37]: Yes, I'll definitely do that!

**Review by laurakalbag — APPROVED** [review]

**Pull request merged**